### PR TITLE
Add update repo secret functionality

### PR DESCRIPTION
**Pull request by nickpagz**
### Changes proposed in this Pull Request

- Add a new command to allow adding or updating action secrets to a site repo.
- Modify the `create-repository` command to automatically add the `GH_BOT_TOKEN` to a new repo (allows the phpcs check to run).

**Some notes:**
- The `update-repository-secret` command is designed to add any secret to the repo. However, in it's current state the command will only add the `GITHUB_API_BOT_SECRETS_TOKEN` token from the **config.json** file to whatever secret name you choose.
- To resolve the above, we need to sort out how to pass a secret. Passing on the command line is impractical, IMO. At the moment, I'm thinking of making it so that the secret first needs to be added to the config.json file and then the code will pick it up from there. 

### Testing instructions

- from your `team51-cli` folder in the terminal, call `git fetch`.
- Switch to the `feature/update-repo-action-secret` branch.

**To test the `update-repository-secret` command:**
- Run a command on any (existing) repo, but create a fake secret name (this can be deleted after).
- Add your fake secret in the `config.json` file in the `team51-cli` folder. Example: `"SECRET_NAME_TEST": "notarealsecret",`
- Command example: `team51 update-repository-secret --repo-slug=nickp-cli-testp --secret-name=SECRET_NAME_TEST`.
- In the repo, go to **Settings > Secrets > Actions**. In the **Repository secrets** box you should see your new secret `SECRET_NAME_TEST`. (Click on **Remove** to delete it).

**To test the  updated `create-repository` command:**
- Create a new test repo using something like `team51 create-repository --repo-slug=my-awesome-project --issue-repo-only`. (Use the "issue-repo-only" flag here to avoid creating a production site and DeployHQ setup).
- Check in https://github.com/orgs/a8cteam51/repositories for your new repo, and go to **Settings > Secrets > Actions**.
- There should be a new token called `GH_BOT_TOKEN`.
- From the main Code section of your repo, switch to the "Develop" branch, click on any file, make a small edit (eg. add some text in the readme file), then commit the change.
- Create a new PR from the commit. 
- The PHPCS should run and complete with the message **All checks have passed**, assuming there are no actual PHP errors.

Delete your test repo (optional).

### CODE REVIEW

**Comment by nickpagz** [comment]
### Update

- `update-repository-secret` will check the config.json file for the secret passed in the command args. Now any secret can be added to a repo.
- Changed flow so we don't `exit()` on verification checks. This allows the `create-repository` command to finish even if some errors occur - such as the secret not existing - since adding the `GH_BOT_TOKEN` is not critical to creating the repo.
- Added some conditionals to check if `GH_BOT_TOKEN` actually exists in config.json, and if not to use the `GITHUB_API_BOT_SECRETS_TOKEN` but still name the secret `GH_BOT_TOKEN`.

**Comment by NickGreen** [comment]
✅  Tested `update-repository-secret` with no params
✅  Tested `update-repository-secret` on repo with no secrets
✅  Tested `update-repository-secret` on repo with existing GH_BOT_TOKEN
✅  Tested `create-repository` to ensure token secret is added
✅  Visually looked through code and new command
[  ]  Ran PHPCS on new code

____


Feedback:

1. Compared to other steps in the process, command line output is a little noisy. I would consider reducing the noise before merging to trunk.

For example, this is output just for the token:
```
Adding GH_BOT_TOKEN for phpcs check.
Secret for GH_BOT_TOKEN found.
Verifying nickg-test-plz-delete does exist in GitHub org.
Repository nickg-test-plz-delete found.
Fetching repository key for nickg-test-plz-delete.
Using repository key 568250167242549743.
Updating repository secret for nickg-test-plz-delete.
Successfully added GH_BOT_TOKEN to (a8cteam51/nickg-test-plz-delete).
```

I would consider silencing all but the first line, unless it fails.

2. PHPCS checks

I found a couple of minor formatting warnings that could be easily fixed:
```
  82 | WARNING | [x] Equals sign not aligned with surrounding assignments; expected 7 spaces but found 3 spaces
     |         |     (Generic.Formatting.MultipleStatementAlignment.NotSameWarning)
  83 | WARNING | [x] Equals sign not aligned with surrounding assignments; expected 5 spaces but found 1 space
     |         |     (Generic.Formatting.MultipleStatementAlignment.NotSameWarning)
```

There are some others that may not be applicable to this, or are legacy with the previous code that we don't want to mess with, and so I'll just ask you take a look at what's there and decide if you want to fix the other stuff, or if it shouldn't be.

File: composer.json [hunk 1]
```diff
@@ -4,7 +4,8 @@
         "symfony/filesystem": "^4.3",
         "symfony/process": "^4.3",
         "phpseclib/phpseclib": "^3.0.7",
-        "amphp/parallel-functions": "^1.0"
+        "amphp/parallel-functions": "^1.0",
+        "phpseclib/phpseclib": "^3.0.7"
```
Comment (NickGreen) [inline composer.json:8]: looks like this line ended up in there twice?

`"phpseclib/phpseclib": "^3.0.7"`

**Review by NickGreen — CHANGES_REQUESTED** [review]
Only one minor comment, otherwise LGTM, merge away!

File: composer.json (hunk 1 above)
Comment (nickpagz) [reply composer.json]: Nice catch! Fixed.

**Pull request merged**